From: SeongJae Park <sj@kernel.org>
Date: Fri, 14 Feb 2025 17:23:32 -0800
Subject: [PATCH] ==== make allow filters after reject filters useful ====

mm/damon: make allow filters after reject filters useful and intuitive

DAMOS filters do allow or reject elements of memory for given DAMOS
scheme only if those match the filter criterias.  For elements that
don't match any DAMOS filter, 'allowing' is the default behavior.  This
makes allow-filters that don't have any reject-filter after them
meaningless sources of overhead.  The decision was made to keep the
behavior consistent with that before the introduction of allow-filters.
This, however, makes usage of DAMOS filters confusing and inefficient.
There is an apparently more intuitive behavior for the case.  If there
is no filter at all or the last filter is a reject filter, allowing by
default.

Update the filtering logic to work in the way.  Decide the default
behavior as the opposite of the last isntalled filter.  If it is an
allow filter, reject by default.  If it is a reject filter or no filter
is installed at all, allow by default.

Note that this is changing the old behavior.  But the old behavior for
the problematic filter combination was definitely confusing, inefficient
and anyway useless.  Also, the behavior has relatively recently
introduced.  It is difficult to anticipate any user that depends on the
behavior.  Hence this is not a user-breaking behavior change but an
obvious improvement.

Changes from RFC
(https://lore.kernel.org/20250220193509.36379-1-sj@kernel.org)
- Rebase on latest mm-unstable
- Wordsmith commit messages
- Set default behavior on core layer filtering stage as allow if any ops
  layer filter exists.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_f274e6so | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_f274e6so

diff --git a/hkml_cv_bogus/hkml_cv_bogus_f274e6so b/hkml_cv_bogus/hkml_cv_bogus_f274e6so
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

